lib/mongo/operation/write/update.rb

```diff
@@ -63,6 +64,10 @@ def write_command_op
           Command::Update.new(s)
         end
 
+        def has_array_filters?
+          update[:array_filters] || update[:arrayFilters] || update['arrayFilters'] || update[Operation::ARRAY_FILTERS]
```

Comment:
We only need to check for ``update[Operation::ARRAY_FILTERS]`` (after changing the constant to ``arrayFilters`` because that's the only key at which we'll add the option from Collection::View::Writable and (should add in) BulkWrite.

Regarding, bulk writes, we have to add the array_filters here: 
https://github.com/mongodb/mongo-ruby-driver/blob/master/lib/mongo/bulk_write/transformable.rb#L120
https://github.com/mongodb/mongo-ruby-driver/blob/master/lib/mongo/bulk_write/transformable.rb#L106
https://github.com/mongodb/mongo-ruby-driver/blob/master/lib/mongo/bulk_write/transformable.rb#L92